crates/proto/src/op/message.rs

```diff
@@ -665,12 +667,23 @@ impl Message {
                 continue;
             }
 
-            match record.record_type() {
+            match record.data() {
                 #[cfg(feature = "__dnssec")]
-                RecordType::SIG => sig = MessageSignature::Sig0(record),
+                RData::Update0(RecordType::SIG) | RData::DNSSEC(DNSSECRData::SIG(_)) => {
+                    sig = MessageSignature::Sig0(record)
+                }
                 #[cfg(feature = "__dnssec")]
-                RecordType::TSIG => sig = MessageSignature::Tsig(record),
-                RecordType::OPT => {
+                RData::Update0(RecordType::TSIG) | RData::DNSSEC(DNSSECRData::TSIG(_)) => {
+                    sig = MessageSignature::Tsig(
+                        record
+                            .map(|data| match data {
+                                RData::DNSSEC(DNSSECRData::TSIG(tsig)) => Some(tsig),
+                                _ => None,
+                            })
+                            .unwrap(),
```

Comment:
You wanted comments for these `unwrap()`s, right? How do we we even know this `unwrap()` won't trigger for the `Update0` case? It seems like it should? Do we have test coverage for that?

Reply:
Good point, I think probably the `RData::Update0(RecordType::TSIG) |` and `RData::Update0(RecordType::SIG) |`  parts of the match arms should be removed alongside the changes to the `message.rs` unit tests switching how the record data is mocked.

Reply:
It's still slightly different than what you had because I'm proposing we keep the `RData::Update0(RecordType::OPT) | ` addition below, alongside the `RData::OPT(_)`. That was the crux of what fixed the tests. We can remove that after fixing the broader `Update0` mess.

If we keep the split commit approach for the typing part we'll also need the first `RData::Update0(RecordType::SIG) | ` part temporariy until the mocks are updated fully in the next commit.